audinterface/core/process.py

```diff
@@ -165,6 +177,21 @@ def process_func(signal, _):
         if win_dur is not None and hop_dur is None:
             hop_dur = utils.to_timedelta(win_dur, sampling_rate) / 2
 
+        # figure out if special arguments
+        # to pass to the processing function
+        signature = inspect.signature(process_func)
+        self._process_func_special_args = {
+            'idx': False,
+            'root': False,
+            'file': False,
+        }
+        for key in self._process_func_special_args:
+            if (
+                    key in signature.parameters
+                    and key not in process_func_args
+            ):
+                self._process_func_special_args[key] = True
```

Comment:
As you need to repeat this for each interface and it is not so easy to read, maybe replace it by a helper function?

Then you could write here:

```python
self._process_func_special_args = utils.check_special_args(
    ['idx', 'root', 'file'],
    process_func,
    process_func_args,
)
```

The only downside it that you not easily see then that a dictionary is created.

Reply:
Actually, not every interface, just `Process' and once for `ProcessWithContext`. So I'm not sure if it's really worth to create a helper function for it.

Reply:
OK, and as you can more easily see that we assign dictionaries, we simply stay with the current solution.